### DISCUSSION TITLE
How to get GitHub request limit state?

### DISCUSSION THREAD

**Discussion by sandragryga** [community/community#23857] [New to GitHub]
<p>Hi!<br>
In our scripts, we frequently use the following command:<br>
<code>Preformatted text hub api --paginate -XGET orgs/our_name/actions/runners -f per_page=100</code></p>
<p>We would like to monitor the number of requests we generate, however, we don’t know how to retrieve the information we need. We tested the following queries:</p>
<pre><code class="lang-auto">curl -H “Accept: application/vnd.github.v3+json” https://api.github.com/rate_limit
curl -I https://api.github.com/user/repos
curl -I https://api.github.com/user
curl -I https://api.github.com/org
</code></pre>
<p>None of them, however, returned results related to the above-mentioned command</p>
<p>Thanks for your help,<br>
Sandra</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Ratelimit information is provided in the headers for every API response, no matter which endpoint you use. Check the header fields with names starting with <code>x-ratelimit</code>. Their meanings are described in the docs:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting" target="_blank" rel="noopener">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/5814919/181079731-52b9932c-e533-4681-9838-b5f241fc8730.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting" target="_blank" rel="noopener">Resources in the REST API - GitHub Docs</a></h3>



  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by sandragryga** [comment]
<p>Thanks. It worked :slight_smile:</p>

**Comment by thanosalexandris** [comment]
I have been marked and i get  API rate limit excedeed for user ID XXXXX
Can you help me please?? why it's so hard here to create a ticket?
